aptos-move/e2e-testsuite/src/tests/execution_strategies.rs

```diff
@@ -16,12 +16,25 @@ use aptos_language_e2e_tests::{
         types::Executor,
     },
 };
-use aptos_types::{transaction::SignedTransaction, vm_status::VMStatus};
+use aptos_types::{
+    transaction::{ExecutionStatus, SignedTransaction, TransactionStatus},
+    vm_status::VMStatus,
+};
 
 fn txn(seq_num: u64) -> SignedTransaction {
     let account = Account::new();
     let aptos_root = Account::new_aptos_root();
-    create_account_txn(&aptos_root, &account, seq_num + 1)
+    create_account_txn(&aptos_root, &account, seq_num)
+}
+
+fn execute_and_assert_success<T>(exec: &mut impl Executor<Txn = T>, block: Vec<T>) {
+    let output = exec.execute_block(block).unwrap();
```

Comment:
if `exec` has multiple executors, is the output scale to number of executors? if not, this function is not suitable for multiple executors

Reply:
e.g. if 2 partitioned executors, for a block of 10 txn, there should be 10*2 execution results

Reply:
What do you mean by multiple executors? `exec` here is a trait that allows to return a vector of transaction outputs.

Reply:
I meant for below case `MultiExecutor`
```
 let mut exec = MultiExecutor::<AnnotatedTransaction, VMStatus>::new();
        exec.add_executor(GuidedExecutor::new(PartitionedGuidedStrategy));
        exec.add_executor(GuidedExecutor::new(UnPartitionedGuidedStrategy));
        execute_and_assert_success(&mut exec, block1);
```

Reply:
I added an assert for the output size, but that would be unaffected by number of executors.

Regarding the fact that each executor is used to execute the block, I think that should be a separate test and PR (since it's also orthogonal to fixing sequence number / success issue).

Reply:
it is not up to me to decide if a separate pr or not :)
I would like to  comment that tests should assert the right verdicts, this is actually the root cause that it was not discovered before. 
The the right number of output txn is an important verdict to check, otherwise, how do we know if `MultiExecutor` do the right work?
Can's the check function get how many executor inside it, calc the expected number of output txns?

Reply:
I think multi-executor internally checks outputs from different executors against each other, and the size of the output stays the same.